### ISSUE TITLE
Both `typing.ByteString` and `typing.Text` are deprecated and slated for removal in Python 3.14

### ISSUE THREAD

**Issue by flying-sheep** [graphql-python/graphql-core#211]
Since 3.12, Python raises a DeprecationWarning for `typing.ByteString`

https://github.com/graphql-python/graphql-core/blob/0c93b8452eed38d4f800c7e71cf6f3f3758cd1c6/src/graphql/pyutils/is_iterable.py#L2
https://github.com/graphql-python/graphql-core/blob/0c93b8452eed38d4f800c7e71cf6f3f3758cd1c6/src/graphql/pyutils/is_iterable.py#L22

[It says](https://docs.python.org/3.12/library/typing.html#typing.ByteString):

> This type represents the types [bytes](https://docs.python.org/3.12/library/stdtypes.html#bytes), [bytearray](https://docs.python.org/3.12/library/stdtypes.html#bytearray), and [memoryview](https://docs.python.org/3.12/library/stdtypes.html#memoryview) of byte sequences.
> 
> Deprecated since version 3.9, will be removed in version 3.14: Prefer [collections.abc.Buffer](https://docs.python.org/3.12/library/collections.abc.html#collections.abc.Buffer), or a union like bytes | bytearray | memoryview.

Since `typing.Text` is a deprecated (since 3.11) alias to `str`, both should be removed here at the same time.

I’d do a PR, but I’m not sure if it should be

```py
 not_iterable_types: tuple[type, ...] = (bytes, Mapping, str)
```

or

```py
 not_iterable_types: tuple[type, ...] = (Buffer, Mapping, str)
```

**Comment by Cito** [comment]
Thanks. Will care about that. Still waiting for a free moment to work on graphql-core again.

**Comment by srittau** [comment]
I think these issues have been fixed in HEAD.

**Comment by Cito** [comment]
Yes, should be fixed in 3.3.0a5.
